### DISCUSSION TITLE
how to pass a dictionary to another ui.page?

### DISCUSSION THREAD

**Discussion by ghost** [zauberzeug/nicegui#2324] [Q&A] [answered]
### Question

I have a page to input some values, and then a button to go to another page. I like to pass the values to the other page from the first page. Example code below:

```py
@ui.page('/otherpage')
def otherpage(payload): # payload is a dictionary of values from main page
    ui.label(payload)

@ui.page('/main')
def main():
    new = {}
    name = ui.input(label="name:").bind_value_to(new, 'name')
    address = ui.textarea(label="address:").bind_value_to(new, 'address')
    age = ui.number(label='age:').bind_value_to(new,'age')
    ui.button("submit", on_click= lambda: ui.open('otherpage')) # how to pass dict new?
```
I have looked through the docs and googled for the solution, but am not able to find anything easy to understand. I am comfortable with python, but very new to web development (which is why I like nicegui very much).  Greatly appreciate if anyone can help me here. 
p.s. Passing path parameters is not suitable for me, as in my real use case, the dictionary is much bigger.

**Comment by rodja** [comment]
You can pass the parameters "classical" via url (eg. `ui.open('/otherpage?name=...`) or use `app.storage.user`.

**Comment by falkoschindler** [comment, marked as answer]
Here is an example using `app.storage.user`:

```py
from nicegui import app, ui

@ui.page('/otherpage')
def otherpage():
    data = app.storage.user['data']
    ui.json_editor({'content': {'json': data}})

@ui.page('/')
def main():
    new = {}
    ui.input('name:').bind_value_to(new, 'name')
    ui.textarea('address:').bind_value_to(new, 'address')
    ui.number('age:').bind_value_to(new, 'age')

    def open_other_page():
        app.storage.user.update(data=new)
        ui.open('otherpage')
    ui.button('submit', on_click=open_other_page)

ui.run(storage_secret='secret')
```

**Reply by ghost** [reply]
This solution works great! Thank you so much! love this framework more each day...

**Comment by NebularNerd** [comment]
For those looking at this in October 2024, replace `ui.open` with `ui.navigate.to` to match new codebase changes
